Core/Core/Submissions/Submission.swift

```diff
@@ -333,21 +333,81 @@ extension Submission {
         return excused == true || (score != nil && workflowState == .graded)
     }
 
-    /// Returns the appropriate status text.
+    public var status: SubmissionStatus {
```

Comment:
It may worth to keep this unchanged code in it's original place, which would make the diff clearer and keep the original Authors info intact.